x/denommetadata/types/packet_metadata.go

```diff
@@ -0,0 +1,27 @@
+package types
+
+import (
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// MemoData represents the structure of the memo with user and hub metadata
+type MemoData struct {
+	TransferInject *TransferInject `json:"transferinject,omitempty"`
+}
+
+type TransferInject struct {
+	DenomMetadata *types.Metadata `json:"denom_metadata,omitempty"`
+}
+
+func (p TransferInject) ValidateBasic() error {
+	return p.DenomMetadata.Validate()
+}
+
+func ParsePacketMetadata(input string) *TransferInject {
+	bz := []byte(input)
+	var memo MemoData
+	_ = json.Unmarshal(bz, &memo) // we don't care about the error
```

Comment:
why not to return the err as well?

Reply:
I think we don't need it. We only care about extracting the memo, if it's there